CMakeLists.txt

```
@@ -0,0 +1,103 @@
+cmake_minimum_required(VERSION 3.22)
+
+include(FetchContent)
+
+project(ClpIrV1Decoder
+    LANGUAGES CXX C
+)
+
+# Enforce "Unix Makefiles" because "Ninja" does not work with the Emscripten toolchain on CXX_20 sources.
+if(NOT CMAKE_GENERATOR MATCHES "Unix Makefiles")
+    message(FATAL_ERROR "This project is intended to be built with Unix Makefiles on a Unix system.")
+endif()
+
+# Enable compile commands by default if the generator supports it.
+if (NOT CMAKE_EXPORT_COMPILE_COMMANDS)
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE BOOL
+        "Enable/Disable output of compile commands during generation." FORCE)
+endif()
+
+# Set the default build type to Release if not specified
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "Setting build type to 'Release' as none was specified.")
```

Let's add a variable to deduplicate (a general practice to avoid manual errors as well).
```suggestion
    set(default_build_type "Release")
    message(STATUS "No build type specified. Setting to '${default_build_type}'.")
```